### DISCUSSION TITLE
Validation for node locked licensing model

### DISCUSSION THREAD

**Discussion by Ponsanthini0708** [Labs64/NetLicensing-Community#140] [Questions] [answered]
I am trying do the validation for the node locked licensing model + subscription licensing model through a single licensee key. 
From the documentation , I can see that for nodelocked model we have to pass the nodeSecret which is created along with the product module number. But in tha validate API, I can see only three parameters which we can able to pass as below:

ValidationResult vRes = LicenseeService::validate(ctx, licenseeNumber, vParams);

How to validate the nodelocked license module by passing the nodeSecret and the productmoduleNumber?

**Comment by r-brown** [comment, marked as answer]
Hello @Ponsanthini0708 
You can provide additional validation parameters for the product module using `setProductModuleValidationParameters` method:

```
    ValidationParameters vParams = ValidationParameters();
    vParams.setProductModuleValidationParameters(productModuleNumber, "paramKey", "paramValue");
    ValidationResult vres = LicenseeService::validate(ctx, licenseeNumber, vParams);
```

See also C++ demo at https://github.com/Labs64/NetLicensingClient-cpp/blob/master/client_demo/netlicensing_client_demo.cc#L372

**Comment by r-brown** [comment]
Issue moved from the [Labs64/NetLicensingClient-cpp](https://github.com/Labs64/NetLicensingClient-cpp)

See also related comment here https://github.com/Labs64/NetLicensingClient-cpp/issues/8#issuecomment-2103881305

**Reply by Ponsanthini0708** [reply]
Thanks for the response.Can you please tell me how to pass the nodeSecret? Because I can also see that the setLicenseeSecret() API is deprecated.

**Reply by r-brown** [reply]
As outlined above, you can use ValidationParameters object to pass parameters as needed:
```
    vParams.setProductModuleValidationParameters(productModuleNumber, "nodeSecret", "device-fingerprint");
```
